enterprise_access/apps/content_assignments/management/commands/automatically_expire_assignments.py

```diff
@@ -0,0 +1,131 @@
+
+    def fetch_enterprise_customer_admin_emails(self, enterprise_customer_uuid):
+        """ Fetch enterprise customer admin emails """
+        lms_client = LmsApiClient()
+        enterprise_customer_data = lms_client.get_enterprise_customer_data(enterprise_customer_uuid)
+        admin_emails = [user['email'] for user in enterprise_customer_data['admin_users']]
+        return admin_emails
+
+    def handle(self, *args, **options):
+        dry_run = options['dry_run']
+
+        log_prefix = '[AUTOMATICALLY_EXPIRE_ASSIGNMENTS]'
+        if dry_run:
+            log_prefix = '[DRY_RUN]'
+
+        sleep_duration = 5  # number of seconds to sleep during batches
+
+        current_date = timezone.now()
+
+        for assignment_configuration in AssignmentConfiguration.objects.filter(active=True):
+            logger.info(
+                '%s Processing Assignment Configuration. UUID: [%s]',
+                log_prefix,
+                assignment_configuration.uuid
+            )
+
+            subsidy_access_policy = assignment_configuration.subsidy_access_policy
+            enterprise_catalog_uuid = subsidy_access_policy.catalog_uuid
+            subsidy_expiration_datetime = subsidy_access_policy.subsidy_expiration_datetime
+
+            admin_emails = self.fetch_enterprise_customer_admin_emails(
+                assignment_configuration.enterprise_customer_uuid
+            )
+
+            allocated_assignments = assignment_configuration.assignments.filter(
+                state=LearnerContentAssignmentStateChoices.ALLOCATED
+            )
+
+            paginator = Paginator(allocated_assignments, 100)
+            for page_number in paginator.page_range:
+                assignments = paginator.page(page_number)
+
+                content_metadata_for_assignments = get_content_metadata_for_assignments(
+                    enterprise_catalog_uuid,
+                    assignments
+                )
+
+                for assignment in assignments:
+                    content_metadata = content_metadata_for_assignments.get(assignment.content_key, {})
+                    enrollment_end_date = content_metadata.get('normalized_metadata', {}).get('enroll_by_date')
+
+                    logger.info(
+                        '%s AssignmentUUID: [%s], ContentKey: [%s], AssignmentExpiry: [%s], EnrollmentEnd: [%s], SubsidyExpiry: [%s]',  # nopep8 pylint: disable=line-too-long
+                        log_prefix,
+                        assignment.uuid,
+                        assignment.content_key,
+                        assignment.created + timezone.timedelta(days=90),
+                        enrollment_end_date,
+                        subsidy_expiration_datetime
+                    )
+
+                    expired_assignment_uuid = None
+                    # just for logging
+                    assignment_expiry_reason = None
+
+                    if current_date > (assignment.created + timezone.timedelta(days=90)):
```

Comment:
I would move your `current_date = timezone.now()` definition to the line right before this - a non-trivial amount of time may pass during this for loops.